### PR TITLE
Update build to use sbt 1.0 (also enables mima)

### PR DESCRIPTION
**Pull request by marcospereira**
Also, update dependencies. There are a lot of code format changes due to Scalariform update.

### CODE REVIEW

**Comment by hirofumi** [comment]
I noticed the following:

* The test failure may be caused by https://github.com/travis-ci/travis-ci/issues/8836.
   * [Replacing](https://github.com/hirofumi/scalatestplus-play/compare/3fa2e0931ea897079c6d1573fccf5a750f2c965c...43c350e335d21045f2ff8e132ee8e2e4be73dc86) `sudo: false` with `sudo: required` seems to work as a workaround.
   * Or [`--no-sandbox` flag](https://docs.travis-ci.com/user/chrome#Sandboxing) may work.
* Maybe [Scala versions of .travis.yml](https://github.com/marcospereira/scalatestplus-play/blob/a0bddb66ee22c69335dfadceab6845da6f431eb4/.travis.yml#L3-L4) can be updated (along with [those of interplay]( https://github.com/playframework/interplay/blob/1.3.15/src/main/scala/interplay/PlayBuildBase.scala#L20-L21)).

Sorry if I bothered you.

**Comment by marcospereira** [comment]
@hirofumi, thanks for your suggestions. I will try them now. :-)

**Comment by lightbend-cla-validator** [comment]
Hi @Dasiu,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ignasi35** [comment]
Sorry @Dasiu. I messed up rebasing and `--force` pushing and your GitHub handle appeared on this PR by mistake triggering the CLA Validator. You are not required to sign the CLA for this PR.

**Comment by ignasi35** [comment]
@marcospereira, I messed a rebase and ended up squashing the original commits you had pushed into 4e0b312

**Comment by ignasi35** [comment]
Redid the rebase.

The upgrade of scalariform and mima surfaced more code changes and a couple of mima issues (introduced in #136)

**Comment by ignasi35** [comment]
(I did two mistakes trying to rebase this old PR on top of current `master`).

The PR (https://github.com/playframework/scalatestplus-play/pull/110) was created from `master` a long time ago.

Since late Jan 2019, `master` and `4.0.x` diverged with some PRs not back/forward-ported.

Yesterday I rebased the commits from the PR on top of `4.0.x` and today I rebased the branch from `4.0.x` to `master`. On my rebase today git not only took the commits from the PR but also the commits in `4.0.x` that had not been forward-ported to `master` (e.g. https://github.com/playframework/scalatestplus-play/pull/110/commits/0022f6b54ea0bf7dee1aabda032a6d73ccff4c19 which enables MiMa).

Summing up: 
 - the PR (as it is now) is not just bumping sbt to 1.0 but also forward porting #131 and #134

**Comment by ignasi35** [comment]
TIL: GH lists the commits on the PR ordering by date of commit (not position in the git tree)

File: .travis.yml [hunk 1]
```diff
@@ -59,3 +59,14 @@ notifications:
...
+addons:
+  firefox: "latest"
+  chrome: stable
+  apt:
+    packages:
+    - unzip
+    - dbus-x11
+  sauce_connect:
+    username: will_sargent
+    access_key: d809febf-5037-4857-aefb-ac652b4c5315
```
Comment (renatocaval) [inline .travis.yml]: From where is this coming and why the commit message here is "Add oracle jdk 9"?
